src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -728,7 +728,7 @@ public ImmutableArray<T> Replace(T oldValue, T newValue)
         public ImmutableArray<T> Replace(T oldValue, T newValue, IEqualityComparer<T> equalityComparer)
         {
             var self = this;
-            int index = self.IndexOf(oldValue, equalityComparer);
+            int index = self.IndexOf(oldValue, 0, self.Length, equalityComparer);
```

Comment:
Well, I believe the thing I did with ImmutableExtensions.cs is a non-starter, but more of a question of #ifs to remove bits of the library code I can't really have, or partial classes, to allow chunks of code to be omitted. I believe both of those approaches avoids meaningful code duplication.